Framework/Algorithms/src/ChopData.cpp

```diff
@@ -136,12 +134,12 @@ void ChopData::exec() {
       PARALLEL_START_INTERUPT_REGION;
       for (size_t k = 0; k < nbins; k++) {
         size_t oldbin = indexLow + k;
-        workspace->dataY(j)[k] = inputWS->readY(j)[oldbin];
-        workspace->dataE(j)[k] = inputWS->readE(j)[oldbin];
-        workspace->dataX(j)[k] = inputWS->readX(j)[oldbin] - stepDiff;
+        workspace->mutableY(j)[k] = inputWS->y(j)[oldbin];
+        workspace->mutableE(j)[k] = inputWS->e(j)[oldbin];
+        workspace->mutableX(j)[k] = inputWS->x(j)[oldbin] - stepDiff;
       }
-      workspace->dataX(j)[nbins] =
-          inputWS->readX(j)[indexLow + nbins] - stepDiff;
+      workspace->mutableX(j)[nbins] =
+          inputWS->x(j)[indexLow + nbins] - stepDiff;
```

Comment:
Maybe this whole block can be written nicer with iterators? You can avoid the loops an the special treatment of the last bin edge.